andela/travel_tool_back
# #160370654 Enable Managers to See Full Details of Multi-city Requests

#### What does this PR do?
- Enables managers to see full details of trips within a multi-city travel request for approval

#### Description of Task to be completed?
Managers need to see full details of the trips within each request so that there is sufficient information for approval decisions

#### How should this be manually tested?
- Clone this repo - `git clone https://github.com/andela/travel_tool_back.git`
- Change into the repo directory - `cd travel_tool_back`
- Check out this branch - `git checkout ft-approve-multicity-requests-160370654`
- Install dependencies and start the server - `make start`
- Using Postman:
  - Set your token in the Authorization Header - you may need to get an authorization token by login in with a browser and retrieving the token from the browser's developer tools in the Application tab, Session cookies section
  - Send a GET request to `http://localhost:5000/api/v1/requests/<reqeustId>` to retrieve the details of a request:
    - Checkout background context below to see how you can get a testable `requestId`
     

#### Any background context you want to provide?
To get details of a request, you'll need a `requestId`. This is autogenerated in the backend when a new request is created.
- Use Postman to send a POST request to http://localhost:5000/api/v1/requests/ with the data in [this](https://gist.github.com/johnmutuma5/45cd66d793c52319bde97731be994ec5) public gist as the request body
Remember to change the "manager" value in the .json to be your name as it appears in your google account
- Once you get a success response, you can locate a `requestId` within the response body and utilize it in the testing section above

#### What are the relevant pivotal tracker stories?
[#160370654](https://www.pivotaltracker.com/story/show/160370654)

#### Screenshots (if appropriate)
<img width="1144" alt="screen shot 2018-09-17 at 10 19 55 am" src="https://user-images.githubusercontent.com/23254512/45610062-49f32400-ba63-11e8-8cb0-c57dff747e58.png">

#### Questions:
